src/nanoarrow/metadata.c

```diff
@@ -114,8 +114,156 @@ ArrowErrorCode ArrowMetadataGetValue(const char* metadata, const char* key,
   return NANOARROW_OK;
 }
 
+ArrowErrorCode ArrowMetadataGetValue(const char* metadata, const char* key,
+                                     const char* default_value,
+                                     struct ArrowStringView* value_out) {
+  struct ArrowStringView key_view = {key, strlen(key)};
+  return ArrowMetadataGetValueView(metadata, &key_view, default_value, value_out);
+}
+
 char ArrowMetadataHasKey(const char* metadata, const char* key) {
   struct ArrowStringView value;
   ArrowMetadataGetValue(metadata, key, NULL, &value);
   return value.data != NULL;
 }
+
+ArrowErrorCode ArrowMetadataBuilderInit(struct ArrowBuffer* buffer,
+                                        const char* metadata) {
+  ArrowBufferInit(buffer);
+  int result = ArrowBufferAppend(buffer, metadata, ArrowMetadataSizeOf(metadata));
+  if (result != NANOARROW_OK) {
+    return result;
+  }
+
+  return NANOARROW_OK;
+}
+
+ArrowErrorCode ArrowMetadataBuilderAppendView(struct ArrowBuffer* buffer,
+                                              struct ArrowStringView* key,
+                                              struct ArrowStringView* value) {
+  if (value == NULL) {
+    return NANOARROW_OK;
+  }
```

Comment:
Hmm, to me it's a little weird to accept NULL as the value and then just do nothing with it. If we just considered `append(key, NULL)` to be an error, we could drop this, and then we could pass the views by value instead of indirecting through a pointer